src/operation/union/CascadedPolygonUnion.cpp

```diff
@@ -94,13 +97,58 @@ CascadedPolygonUnion::Union()
     // TODO avoid creating this vector and run binaryUnion off the iterators directly
     std::vector<const geom::Geometry*> geoms(index.items().begin(), index.items().end());
 
-    return binaryUnion(geoms, 0, geoms.size());
+// #ifdef GEOS_OPENMP
+    return pairUnion(geoms);
+// #else
+//     return binaryUnion(geoms, 0, geoms.size());
+// #endif
 }
 
+#ifdef GEOS_OPENMP
+#include <omp.h>
+#endif
 
+/* private */
 std::unique_ptr<geom::Geometry>
-CascadedPolygonUnion::binaryUnion(const std::vector<const geom::Geometry*> & geoms,
-                                  std::size_t start, std::size_t end)
+CascadedPolygonUnion::pairUnion(
+    std::vector<const geom::Geometry*>& inply) const
+{
+    int sz = (int)(inply.size());
+    int nsz = (sz / 2) + (sz % 2);
+    std::vector<std::unique_ptr<geom::Geometry>> outply((std::size_t)nsz);
+
+#pragma omp parallel for
+    for (int i = 0; i < sz; i += 2) {
+        /* Work backwards to preserve an old regression result */
```

Comment:
Hum I don't see this backward effect in the binaryUnion() implementation. At least not in an obvious way